src/utils/open-browser-cra/index.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict'
+
+const { execSync } = require('child_process')
+const { join } = require('path')
+const process = require('process')
+
+const chalk = require('chalk')
+const spawn = require('cross-spawn')
+const open = require('open')
+
+// https://github.com/sindresorhus/open#app
+const OSX_CHROME = 'google chrome'
+
+const Actions = Object.freeze({
+  NONE: 0,
+  BROWSER: 1,
+  SCRIPT: 2,
+})
+
+const getBrowserEnv = function () {
+  // Attempt to honor this environment variable.
+  // It is specific to the operating system.
+  // See https://github.com/sindresorhus/open#app for documentation.
+  const value = process.env.BROWSER
+  const args = process.env.BROWSER_ARGS ? process.env.BROWSER_ARGS.split(' ') : []
+  let action
+  if (!value) {
+    // Default.
+    action = Actions.BROWSER
+  } else if (value.toLowerCase().endsWith('.js')) {
+    action = Actions.SCRIPT
+  } else if (value.toLowerCase() === 'none') {
+    action = Actions.NONE
+  } else {
+    action = Actions.BROWSER
+  }
+  return { action, value, args }
+}
+
+const executeNodeScript = function (scriptPath, url) {
+  const extraArgs = process.argv.slice(2)
+  const child = spawn(process.execPath, [scriptPath, ...extraArgs, url], {
+    stdio: 'inherit',
+  })
+
+  child.on('close', (code) => {
+    if (code !== 0) {
+      console.log()
+      console.log(chalk.red('The script specified as BROWSER environment variable failed.'))
+      console.log(`${chalk.cyan(scriptPath)} exited with code ${code} .`)
+      console.log()
+    }
+  })
+
+  return true
+}
+
+const startBrowserProcess = function (browser, url, args) {
+  // If we're on OS X, the user hasn't specifically
+  // requested a different browser, we can try opening
+  // Chrome with AppleScript. This lets us reuse an
+  // existing tab when possible instead of creating a new one.
+  const shouldTryOpenChromiumWithAppleScript =
+    process.platform === 'darwin' && (typeof browser !== 'string' || browser === OSX_CHROME)
+
+  if (shouldTryOpenChromiumWithAppleScript) {
+    // Will use the first open browser found from list
+    const supportedChromiumBrowsers = [
+      'Google Chrome Canary',
+      'Google Chrome',
+      'Microsoft Edge',
+      'Brave Browser',
+      'Vivaldi',
+      'Chromium',
+    ]
+
+    for (const chromiumBrowser of supportedChromiumBrowsers) {
+      try {
+        // Try our best to reuse existing tab
+        // on OSX Chromium-based browser with AppleScript
+        execSync(`ps cax | grep "${chromiumBrowser}"`)
```

Comment:
The code seems to be inconsistent about how processes are being spawned: it uses both `cross-spawn` and `childProcess.execSync()`, while our code uses `execa`. 

`execa` uses `cross-spawn` and adds a few improvements on top of it, while `cross-spawn` uses `childProcess.spawn()`, and adds a few cross-platform improvements on top of it. `childProcess.execSync()` uses `childProcess.spawn()` under the hood.

It seems to me it might be worth it to use `execa` and `async`/`await` instead?